usecases/modules/compare.go

```diff
@@ -0,0 +1,137 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package modules
+
+import (
+	"context"
+
+	"github.com/weaviate/weaviate/entities/additional"
+	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/entities/modulecapabilities"
+	"github.com/weaviate/weaviate/entities/moduletools"
+	"github.com/weaviate/weaviate/entities/schema"
+	"github.com/weaviate/weaviate/entities/search"
+)
+
+func reVectorize(ctx context.Context, cfg moduletools.ClassConfig, mod modulecapabilities.Vectorizer, object *models.Object, class *models.Class, sourceProperties []string, targetVector string, findObjectFn modulecapabilities.FindObjectFn) (bool, models.AdditionalProperties, []float32) {
+	textProps, mediaProps, err := mod.VectorizedProperties(cfg)
+	if err != nil {
+		return true, nil, nil
+	}
+
+	type compareProps struct {
+		Name    string
+		IsArray bool
+	}
+	propsToCmpare := make([]compareProps, 0)
+
+	for _, prop := range class.Properties {
+		if len(prop.DataType) > 1 {
+			continue // multi cref
+		}
+
+		// for named vectors with explicit source properties, skip if not in the list
+		if sourceProperties != nil {
+			found := false
+			for _, sourceProp := range sourceProperties {
+				if prop.Name == sourceProp {
+					found = true
+					break
+				}
+			}
+			if !found {
+				continue
+			}
+		}
+
+		if prop.ModuleConfig != nil {
+			if modConfig, ok := prop.ModuleConfig.(map[string]interface{})[class.Vectorizer]; ok {
```

Comment:
`class.Vectorizer` is relevant only for legacy vector. target vectors use ones configured in `VectorConfig`

Reply:
yep, but if I a not missing something we do not have a skip property in `VectorConfig`

There, you either:
- set the vectorizer to None => no vectorization and won't reach this part of the code
- set the vectorizer to not-None => if you don't want a property to be vectorized you set the `sourceProperties`

Reply:
ok, got it.

I got confused with all the settings that are available now. There is module config on class level (with explicit list of props), module config on property level (with skip) and sourceProperties in vector config.
Not sure what precedence is or if old one are deprecated/ignored for target vectors.
We need to make sure though that the same rules are applied to comparison and vectorization, so the first one does not give invalid `needsRevectorization` result when vectorization would in fact produce different vector.